src/components/ContactForm.js

```diff
@@ -0,0 +1,73 @@
+import React, {useState} from 'react';
+import { Link } from 'react-router-dom';
+import { v4 as uuidv4 } from 'uuid';
+import {formatPhoneNumber} from '../util'; 
+
+const ContactForm = (props) => {
+  //state
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [phone, setPhone] = useState(''); 
+  const [picture, setPicture] = useState(''); 
+
+  const phoneInputHandler = (e) => {
+    const formattedPhoneNumber = formatPhoneNumber(e.target.value);
+    setPhone(formattedPhoneNumber); 
+  };
+
+  const submitContactHandler = (e) => {
+    e.preventDefault(); 
+
+    if(name === '' || email === '' || phone === '' || picture === '') {
+      alert('All fields required.');
+      return; 
+    };
+
+    props.addContact({
+      name: name,
+      email: email,
+      phone_number: phone,
+      image_url: picture,
+      id: uuidv4()
+    });
+
+    props.history.push("/")
```

Comment:
There's a hook for history now so you don't have to pass it! https://reactrouter.com/web/api/Hooks